kani-compiler/src/codegen_cprover_gotoc/compiler_interface.rs

```diff
@@ -115,6 +122,13 @@ impl GotocCodegenBackend {
             call_graph,
         );
 
+        // Re-collect reachable items after global transformations were applied. This is necessary
+        // since global pass could add extra calls to instrumentation.
```

Comment:
Yes, for now we do, since we only have one global analysis. Perhaps we need to think about a good way to handle that in the future if we have multiple global analyses

Reply:
I think this has more to do with the logic being brought up than the number of global analysis, no?

Reply:
I think at least it is reasonable to mandate that the instrumentation code added in a global pass should not impact its own logic. Arguably, the developer should be aware about this code being added and take it into account when performing the pass.